### PR TITLE
[MRG] (f)gw barycenter solvers new features and cleaning

### PR DESCRIPTION
**Pull request by cedricvincentcuaz**
## Types of changes

- add a new option to (un)regularized (f)gw barycenter solvers : `conv_criterion` taking values into `{"barycenter", "loss"}` so the user can choose convergence criterions either based on the absolute norm variations of the barycenter, or relative loss variations, respectively : The last criterion is most likely meaningful for most applications, plus suit to the ones used in inner (f)gw solvers based on conditional gradient. This feature is also added to the entropic (f)gw barycenter solvers and would come with an overhead of computing the loss at the last iteration of the inner entropic solvers. Whereas for CG inner functions, computing the barycenter loss is performed in O(#samples). 

- Trying to harmonise the behavior across all (f)gw barycenter solvers :
     - Some solvers were updating their convergence criterions at only few epochs to reduce the computational burden of computing norms over the barycenter, but it is really not convenient to analyse the convergence of these solvers.
     - Add `fixed_structure` and `fixed_features` to `entropic_fused_gromov_barycenter`, same than unregularized solver.

- Adapt and improve these functions documentation.

## How has this been tested (if it applies)

- Previous tests are running. Additional tests to do.

## PR checklist

- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: ot/gromov/_bregman.py [hunk 1]
```diff
@@ -945,67 +996,108 @@ def entropic_fused_gromov_barycenters(
...
 
     if log:
         log_ = {}
-        log_['err_feature'] = []
-        log_['err_structure'] = []
-        log_['Ts_iter'] = []
+        if conv_criterion == 'barycenter':
+            log_['err_feature'] = []
+            log_['err_structure'] = []
+            log_['Ts_iter'] = []
```
Comment (cedricvincentcuaz) [inline ot/gromov/_bregman.py:1052]: I am really not convinced that `Ts_iter` is useful to anyone ^^'

File: ot/gromov/_bregman.py [hunk 2]
```diff
@@ -345,8 +345,9 @@ def entropic_gromov_wasserstein2(
 
 def entropic_gromov_barycenters(
         N, Cs, ps=None, p=None, lambdas=None, loss_fun='square_loss',
-        epsilon=0.1, symmetric=True, max_iter=1000, tol=1e-9, warmstartT=False,
-        verbose=False, log=False, init_C=None, random_state=None, **kwargs):
+        epsilon=0.1, symmetric=True, max_iter=1000, tol=1e-9,
+        conv_criterion='barycenter', warmstartT=False, verbose=False,
```
Comment (rflamary) [inline ot/gromov/_bregman.py:349]:
```suggestion
        stop_criterion='barycenter', warmstartT=False, verbose=False,
```

File: ot/gromov/_bregman.py [hunk 3]
```diff
@@ -418,6 +423,9 @@ def entropic_gromov_barycenters(
     if loss_fun not in ('square_loss', 'kl_loss'):
         raise ValueError(f"Unknown `loss_fun='{loss_fun}'`. Use one of: {'square_loss', 'kl_loss'}.")
 
+    if conv_criterion not in ['barycenter', 'loss']:
```
Comment (rflamary) [inline ot/gromov/_bregman.py:426]: stop

File: ot/gromov/_bregman.py [hunk 4]
```diff
@@ -446,45 +454,72 @@ def entropic_gromov_barycenters(
...
     cpt = 0
-    err = 1
-
-    error = []
+    err = 1e15  # either the error on 'barycenter' or 'loss'
 
     if warmstartT:
         T = [None] * S
 
+    if conv_criterion == 'barycenter':
```
Comment (rflamary) [inline ot/gromov/_bregman.py:462]: same here

**Review by rflamary — COMMENTED** [review]
Nice PR. I think we shoumd call it `stop_criterion`instead of conv it is mor eclear to me WDYT?

File: ot/gromov/_bregman.py (hunk 2 above)
Comment (cedricvincentcuaz) [reply ot/gromov/_bregman.py:349]: Yep sounds perfect, thanks :)

File: RELEASES.md [hunk 5]
```diff
@@ -17,6 +17,8 @@
 + Add KL loss to all semi-relaxed (Fused) Gromov-Wasserstein solvers (PR #559)
 + Further upgraded unbalanced OT solvers for more flexibility and future use (PR #551)
 + New API function `ot.solve_sample` for solving OT problems from empirical samples (PR #563)
++ Add `conv_criterion` feature to (un)regularized (f)gw barycenter solvers (PR #578)
```
Comment (rflamary) [inline RELEASES.md:20]:
```suggestion
+ Add `stop_criterion` feature to (un)regularized (f)gw barycenter solvers (PR #578)
```

**Pull request merged**